segmentation/utils/signal_extraction.py

```diff
@@ -1,31 +1,95 @@
+    improves on default_extraction by not distinguishing between pixel expression values
 
     Args:
-        cell_coords: tuples of values representing pixels within one cell
-        image_data: array containing channel counts
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
 
     Returns:
-        np.array: sum of counts for each channel
+        channel_counts (numpy): sum of counts for each channel
     """
 
-    pass
+    # transpose coords so they can be used to index an array
+    cell_coords = cell_coords.T
+    channel_values = image_data.values[tuple(cell_coords)]
+
+    # sum up based on a binary mask that is 1 if the expression value > 0 else 0
+    channel_counts = np.sum(channel_values > 0, axis=0)
+
+    return channel_counts
+
+
+def center_weighting_extraction(cell_coords, image_data):
+    """
+    Extract channel counts by summing over weighted expression values based on distance from center,
+    improves upon default extraction by including a level of certainty/uncertainty
+
+    Args:
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
+
+    Returns:
+        channel_counts (numpy): sum of counts for each channel
+    """
+
+    # create a weighting matrix, we will assign the highest value at the center
+    # and decrease by 1 each layer we move away from it
+    # I'm going to be assume we get square matrices in image_data
+    center = int(image_data.shape[0] / 2)
+
+    # this will ensure we never get zero or negative weighting values in our weight matrix
+    center_weight = center + 1
+
+    # now build the weight matrix using ogrid
+    image_row_index, image_col_index = np.ogrid[:image_data.shape[0], :image_data.shape[1]]
+    weight_matrix = center_weight - np.maximum(np.abs(image_row_index - center), np.abs(image_col_index - center))
+
+    # TODO: the center of even-length arrays is not a single pixel which currently causes
+    # minor signal irregularities at different layers
+    # ex. in a 4 x 4 matrix the center should be:
+    # 0 0 0 0
+    # 0 X X 0
+    # 0 X X 0
+    # 0 0 0 0
+
+    # and not:
+    # 0 0 0 0
+    # 0 0 0 0
+    # 0 0 X 0
+    # 0 0 0 0
+    # as would be defined without the correction by the regular signal weighting algorithm
+    # the easy way to fix this is to use skimage.draw.rectangle
+
+    # center the weight matrix by center_weight
+    weight_matrix = weight_matrix / center_weight
+
+    # transpose coords so they can be used to index an array
+    cell_coords = cell_coords.T
+    channel_values = image_data.values[tuple(cell_coords)]
+    weight_values = np.expand_dims(weight_matrix[tuple(cell_coords)], axis=0)
+
+    # multiply channel_values by weight_values and then sum across channels
+    channel_counts = np.sum(np.multiply(channel_values, weight_values.T), axis=0)
+
```

Comment:
Another way of doing this:
```
# im assuming cell_coords is a num_cells-by-2 array
num_cell_coords = cell_coords.shape[0]

# compute centroid
centroid = np.sum(cell_coords, axis=0)/num_cell_coords

# compute distances (ord=np.inf for box norm, ord=2 for regular euclid time (default))
distances = np.linalg.norm(cell_coords - centroid, axis=1)

# idk how exactly we want to scale the weighting but this is the current version
weights = 1 - (distances/(np.max(distances) + 1))

channel_values = image_data.values[tuple(cell_coords.T)]

# these should be already lined-up correctly
channel_counts = weights.dot(channel_values)
```
might be a few indexing+typos here but conceptually should work